// instagram-private-api/client/v1/feeds/account-followers.js
var _ = require('underscore');
var util = require('util');
var FeedBase = require('./feed-base');

function AccountFollowersFeed(session, accountId, limit) {
    this.accountId = accountId;
    this.limit = limit || 7500;
    // Should be enought for 7500 records
    this.timeout = 10 * 60 * 1000;
    FeedBase.apply(this, arguments);
}
util.inherits(AccountFollowersFeed, FeedBase);

module.exports = AccountFollowersFeed;
var Request = require('../request');
var Account = require('../account');

AccountFollowersFeed.prototype.get = function () {
    var that = this;
    return new Request(that.session)
        .setMethod('GET')
        .setResource('followersFeed', {
            id: that.accountId,
            maxId: that.cursor
        })
        .send()
        .then(function(data) {
            console.log('that cursor:', that.cursor);
            // console.log('data:', data);
            // console.log('that:', that);
            that.moreAvailable = !!data.next_max_id;
            if (that.moreAvailable) {
                that.setCursor(data.next_max_id);
            }
            return _.map(data.users, function (user) {
                return new Account(that.session, user);
            });
        })
};

AccountFollowersFeed.prototype.resume = function (cursor) {
    var that = this;
    return new Request(that.session)
        .setMethod('GET')
        .setResource('followersFeed', {
            id: that.accountId,
            maxId: cursor
        })
        .send()
        .then(data => {
            that.moreAvailable = !!data.next_max_id;
            if (that.moreAvailable) {
                that.setCursor(data.next_max_id);
            }
            return _.map(data.users, user => {
                return new Account(that.session, user);
            })
        })
}
